Icon.gd

```diff
@@ -4,6 +4,7 @@ const Waypoint = preload("res://waypoint.gd")
 
 var texture_path
 var waypoint: Waypoint.Icon
+var category: TreeItem
```

Comment:
Why do we need to store the TreeItem for all the Icons and Routes? It seems a little messy to just be giving the markers all these random references to things.

Reply:
So before, we used we stored the name of the category that a node belonged to in `node.waypoint.get_category().get_name()`. This was referenced later so that we could check against visibility of the category TreeItem.  Since we changed it so there is no longer a category object in those messages, I need something to be able to find the category from the node. Protobuf does not have a `.parent()` type function that i can use. My first thought was just to pass the name but if we have a way to edit the name in the program, it wouldn't propagate.

Reply:
I understand this logic, but it still feels weird to have the internals of a marker 3d node so tightly coupled with the internal workings of the tree selector UI. Additionally we put a bunch of effort into nesting the proto data, why are we not using that here, just calling hide/show on a category and recursing down through the children to determine which ones should be hidden or shown.

That might be a lot more work then this but I feel like this goes a little to far in the wrong direction. Because it means that when we inevitably change the category tree selector UI we will have to go through and update the marker nodes too, aka undo the work of this PR. Sometimes that would be acceptable if things were being actively transitioned but at this point we should just do it better as we probably wont be changing it again before the xml_parser merge.